pkg/meta/redis.go

```diff
@@ -325,23 +358,53 @@ func (r *redisMeta) StatFS(ctx Context, totalspace, availspace, iused, iavail *u
 }
 
 func (r *redisMeta) Lookup(ctx Context, parent Ino, name string, inode *Ino, attr *Attr) syscall.Errno {
-	buf, err := r.rdb.HGet(c, r.entryKey(parent), name).Bytes()
-	if err != nil {
-		return errno(err)
+	var foundIno Ino
+	var encodedAttr []byte
+	var err error
+
+	entryKey := r.entryKey(parent)
+	if len(shaLookup) > 0 {
+		var res interface{}
+		res, err = r.rdb.EvalSha(c, shaLookup, []string{entryKey, name}).Result()
+		if err != nil {
+			return errno(err)
+		}
+		invalidScriptRes := fmt.Errorf("invalid script result: %v", res)
```

Comment:
It's better to leave this out of the common path (construct an error when needed)